CONTRIBUTING.md

```diff
@@ -5,20 +5,91 @@
+2. `fix`.
+Everything that doesn't fit in `not-code`, but doesn't change set of exported definitions and their types or semantics.
+Examples: fix compatibility with a new version of a library/compiler, improve performance of some function (keeping its semantics intact).
+Note: despite the name, issues of this type do not necessary correspond to bug fixes.
+3. `feature`.
+Modifications which make set of exported definitions a superset of the old set of exported definitions.
+I. e. all existing definitions are left intact, only something new is added.
+Example: add a new function, re-export a type class, add a new instance of some type class.
+4. `breaking`.
+Basically all other changes.
+Examples: remove of a function, rename a type class, change arguments order of a function.
+
+Each issue should belong to only one category.
+If an issue involves multiple changes of different types, it should be split into smaller issues
+
+For each category we have a label on GitHub, these labels start with the `type:` prefix.
+There is also the `type:unclear` label for issues which don't have a list of specific actions to be done.
+
+Processing of an issue depends on its category. Specifically:
+1. A `not-code` issue can be resolved at any time.
+PRs can be merged whenever GitHub allows it.
+Minimal number of approvals is set to 2 in GitHub settings.
+2. Rules for the `fix` issues are the same as for `not-code` with an additional rule that PRs for such issues should increase the last component in the version and this version should be uploaded to Hackage as soon as the PR is merged (also new tag and release are created).
+3. Work on the `feature` issues is organized in milestones.
+Each milestone corresponds to a release which increases at least the third (`feature`) version component.
+Scope of each milestone is discussed in issues and is set by maintainers.
+In some cases a special issue can be created to discuss the scope of a particular milestone.
+A `feature` issue is added to a milestone if such addition is approved by at least 3 people (by explicitly stating that in GH comments).
+At least 2 of them must be from Serokell.
+However, if there are 5 people (from which at least 3 are from Serokell) who disapprove this issue, it can't be added to a milestone.
+A new `feature` release can happen only after at least 2 weeks have passed after last release (except `fix` releases).
+PR can be merged only if the corresponding `feature` issue is in the upcoming milestone.
+4. Work on the `breaking` issues is organized in almost the same way.
+However, 4 approvals (at least 3 from Serokell) are necessary for inclusion into a milestone.
+If an issue is about removal of something (`x`), it should be done in two steps.
+The first step is deprecation of `x` which happens in a `breaking` release.
+The second step is deprecation of `x` which happens in another `breaking` release after deprecation of `x`.
+There can't be more than one breaking release in less than 5 weeks.
```

Comment:
What about disapprovals? I think those from serokell should become quickly fatal comparing to `feature` case.

Reply:
Updated slightly (now 2 disses from SRK is enough to block an issue).